zephyr/lc3_decoder.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022 Bose Corporation All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "Lc3Decoder.hpp"
+#include "Lc3Config.hpp"
+#include "lc3_decoder.h"
+#include <cstddef>
+
+Lc3Decoder_t Lc3Decoder_create(uint16_t Fs_, lc3config_frame_duration_t N_ms_)
+{
+    Lc3Decoder* inst;
+    Lc3Config::FrameDuration duration;
+    
+    switch(N_ms_)
+    {
+    case lc3config_duration_7p5ms:
+        duration = Lc3Config::FrameDuration::d7p5ms;
+        break;
+    case lc3config_duration_10ms:
+        duration = Lc3Config::FrameDuration::d10ms;
+        break;
+    default:
+        /* Not all compilers understand that the enum only have two entries
+         * and therefore warns about potential uninitialized value in duration.
+         * This code is dead code. */
```

Comment:
If this is true, then perhaps an __ASSERT or something is better than returning a valid value.

Reply:
Agree - I have changed to return NULL even though it should never happen. Only way this can happen is if you typecast an invalid value into the enum type.
As this module is compiled as a library there is no concept of calling into the zephyr assert system (for example to provide text output), hence I'm not sure we want to add a dependency to an assert function.